# Proyecto 2 Synergy Logistics


<img src=https://images.freeimages.com/images/large-previews/3c9/contenair-ship-1-1238009.jpg  width="900" height="600" >


## Juan Carlos Herrera Pro
## Emtech Institute

## Indice

### 1. Introducción
### 2. Opción Sugeríada como base de la estrategia de Synergy logistics
### 3. Conclusión

## 1. Introducción

La Dirección de Synergy Logistics ha solicitado al equipo operativo, realizar una
propuesta que permita enfocar las prioridades de la estrategia operativa 2021; para
ello, se plantea analizar la viabilidad de 3 opciones de enfoque: rutas de importación
y exportación, medio de transporte utilizado y valor total de importaciones y
exportaciones. 
Como data analyst tu deber es ayudar a encontrar la mejor opción. 
Dentro de las cuales tendremos las siguientes: 

Opción 1) Rutas de importación y exportación. Synergy logistics está
considerando la posibilidad de enfocar sus esfuerzos en las 10 rutas más
demandadas. Acorde a los flujos de importación y exportación, ¿cuáles son esas
10 rutas? ¿le conviene implementar esa estrategia? ¿porqué?


Opción 2) Medio de transporte utilizado. ¿Cuáles son los 3 medios de transporte
más importantes para Synergy logistics considerando el valor de las
importaciones y exportaciones? ¿Cuál es medio de transporte que podrían
reducir?


Opción 3) Valor total de importaciones y exportaciones. Si Synergy Logistics
quisiera enfocarse en los países que le generan el 80% del valor de las
exportaciones e importaciones ¿en qué grupo de países debería enfocar sus
esfuerzos?


## 2. Opción Sugeríada como base de la estrategia de Synergy logistics

## Opción 1) 
Rutas de importación y exportación. Synergy logistics está
considerando la posibilidad de enfocar sus esfuerzos en las 10 rutas más
demandadas. Acorde a los flujos de importación y exportación, ¿cuáles son esas
10 rutas? ¿le conviene implementar esa estrategia? ¿porqué?

In [1]:
import csv

In [70]:
with open("synergy_logistics_database.csv", "r") as archivo_csv:
    lector = csv.reader(archivo_csv)
    paises_exportadores_data = []
    paises_importadores_data = []

    i = 0
    for linea in lector:
        if linea[1] == 'Exports': 
            paises_exportadores_data.append([linea[2], linea[3]])
          
        elif linea[1] == 'Imports':    
            paises_importadores_data.append([linea[2], linea[3]])
    
    
    paises_exportadores = []

    for pais in paises_exportadores_data:
        if pais not in paises_exportadores:
            paises_exportadores.append(pais)   
    

    paises_importadores = []

    for pais in paises_importadores_data:
        if pais not in paises_importadores:
            paises_importadores.append(pais)
    
    total_exportaciones = 0
    paises_exportadores_conteo = []
    for pais in paises_exportadores:
        conteo = 0
        for linea in paises_exportadores_data:
            if pais[0] == linea[0] and pais[1] == linea[1]:
                conteo += 1
        total_exportaciones += conteo
        paises_exportadores_conteo.append([conteo, pais])

    paises_exportadores_conteo.sort(reverse=True)
    
    
    total_importaciones = 0
    paises_importadores_conteo = [] 
    for pais in paises_importadores:
        conteo = 0
        for linea in paises_importadores_data:
            if pais[0] == linea[0] and pais[1] == linea[1]:
                conteo += 1
        total_importaciones += conteo
        paises_importadores_conteo.append([conteo, pais])

    paises_importadores_conteo.sort(reverse=True)
    

   

    print("||=====================Top 10 rutas más transitadas===========================||")
    for i  in range(10):
        print(i+1, "La ruta ", paises_exportadores_conteo[i][1], "con", paises_exportadores_conteo[i][0], "exportaciones" )
        
    print("Total exportaciones: ", total_exportaciones)

    print("----------------------------")
    for i  in range(10):
        print(i+1, "La ruta ", paises_importadores_conteo[i][1], "con", paises_importadores_conteo[i][0], "importaciones" )

    print("Total de importaciones: ", total_importaciones)

||=====================Top 10 rutas más transitadas===========================||
1 La ruta  ['South Korea', 'Vietnam'] con 497 exportaciones
2 La ruta  ['Netherlands', 'Belgium'] con 437 exportaciones
3 La ruta  ['USA', 'Netherlands'] con 436 exportaciones
4 La ruta  ['China', 'Mexico'] con 330 exportaciones
5 La ruta  ['Japan', 'Brazil'] con 306 exportaciones
6 La ruta  ['Germany', 'France'] con 299 exportaciones
7 La ruta  ['South Korea', 'Japan'] con 279 exportaciones
8 La ruta  ['Australia', 'Singapore'] con 273 exportaciones
9 La ruta  ['Canada', 'Mexico'] con 261 exportaciones
10 La ruta  ['China', 'Spain'] con 250 exportaciones
Total exportaciones:  15408
----------------------------
1 La ruta  ['Singapore', 'Thailand'] con 273 importaciones
2 La ruta  ['Germany', 'China'] con 233 importaciones
3 La ruta  ['China', 'Japan'] con 210 importaciones
4 La ruta  ['Japan', 'Mexico'] con 206 importaciones
5 La ruta  ['China', 'Thailand'] con 200 importaciones
6 La ruta  ['Malaysia', 'Th

Implementar está estrategía podría parecer interesante, pero el número de total de exportaciones que es de 15,408 es muy poco comparado con el top 10 exportaciones que es de 3,368. Es decir, la empresa unicamente estaría operando sus exportaciones a un 22 %. 
En el caso de las importaciones, esta idea podría ser algo más atractiva, pues de las 3,648 importaciones totales, 1,808 le pertenecen al top 10, practicamente un 50 %. 

Como recomendación del analista de datos, la opción 1 quedaría descartada. 

## Opción 2) 
Medio de transporte utilizado. ¿Cuáles son los 3 medios de transporte
más importantes para Synergy logistics considerando el valor de las
importaciones y exportaciones? ¿Cuál es medio de transporte que podrían
reducir? 

In [61]:
with open("synergy_logistics_database.csv", "r") as archivo_csv:
    lector = csv.reader(archivo_csv)
    
    transportes_exportadores_data = []
    transportes_importadores_data = []

    for linea in lector:
        if linea[1] == 'Exports': 
            transportes_exportadores_data.append(linea[7])
          
        elif linea[1] == 'Imports':    
            transportes_importadores_data.append(linea[7])

    transportes_exportadores = []
    transportes_importadores = []

    for transporte in transportes_exportadores_data:
        if transporte not in transportes_exportadores:
            transportes_exportadores.append(transporte)

    

    for transporte in transportes_importadores_data:
        if transporte not in transportes_importadores:
            transportes_importadores.append(transporte)

In [72]:
with open("synergy_logistics_database.csv", "r") as archivo_csv:

  transportes_exportadores_suma = []
  lector = csv.reader(archivo_csv) 

  precios = []
  for linea in lector:
    if linea[9] != 'total_value': 
      precios.append(linea[9])
  
  

  total_exportaciones = 0
  for transporte in transportes_exportadores:
    suma = 0
    i = 0
    for transporte2 in transportes_exportadores_data:
      if transporte == transporte2:
          suma += int(precios[i]) 
    total_exportaciones += suma
    i += 1
    transportes_exportadores_suma.append([suma, transporte])

  transportes_exportadores_suma.sort(reverse=True)
  print("Valor total de exportaciones ", total_exportaciones)
  print("||=====================Top 3 medios de transporte que aportan más valor (Exportaciones)===========================||")
  for i in range(3):
    print(i+1,"El medio de transporte ", transportes_exportadores_suma[i][1], "reporta un total de", transportes_exportadores_suma[i][0], " en exportaciones")

Total de exportaciones  508464000000
||=====================Top 3 medios de transporte que aportan más valor (Exportaciones)===========================||
1 El medio de transporte  Sea reporta un total de 278322000000  en exportaciones
2 El medio de transporte  Rail reporta un total de 91740000000  en exportaciones
3 El medio de transporte  Road reporta un total de 70917000000  en exportaciones


In [73]:
with open("synergy_logistics_database.csv", "r") as archivo_csv:

  transportes_importadores_suma = []
  lector = csv.reader(archivo_csv) 

  precios = []
  for linea in lector:
    if linea[9] != 'total_value': 
      precios.append(linea[9])
  

  total_importaciones = 0
  for transporte in transportes_importadores:
    suma = 0
    i = 0
    for transporte2 in transportes_importadores_data:
      if transporte == transporte2:
          suma += int(precios[i]) 
    total_importaciones += suma
    i += 1
    transportes_importadores_suma.append([suma, transporte])

  transportes_importadores_suma.sort(reverse=True)
  print("Valor total de importaciones ", total_importaciones)
  print("||=====================Top 3 medios de transporte que aportan más valor (Importaciones)===========================||")
  for i in range(3):
    print(i+1,"El medio de transporte ", transportes_importadores_suma[i][1], "reporta un total de", transportes_importadores_suma[i][0], " en importaciones")

Valor total de importaciones  120384000000
||=====================Top 3 medios de transporte que aportan más valor (Importaciones)===========================||
1 El medio de transporte  Sea reporta un total de 74382000000  en importaciones
2 El medio de transporte  Rail reporta un total de 19833000000  en importaciones
3 El medio de transporte  Road reporta un total de 14817000000  en importaciones


Reducir el medio de transporte menos utilizado, representa que la empresa mueve un total de 440,979,000,000 valor en exportaciones, lo cual es casi el 87% del total de exportaciones recordemos que este es de 508,464,000,000. 

Mientras que para las importaciones el transporte por aire unicamente representa el 10% del valor total de las importaciones. Pues el convinado de Mar, Riel y tierra es de 109,032,000,000. 

Esta es una buena opción, pues de esta forma, no se tendrían que abandonar rutas de exportación e importación. Y las rutas por aire podrían quedar solamente de emergencia. 

## Opción 3) 
Valor total de importaciones y exportaciones. Si Synergy Logistics
quisiera enfocarse en los países que le generan el 80% del valor de las
exportaciones e importaciones ¿en qué grupo de países debería enfocar sus
esfuerzos?


In [57]:
with open("synergy_logistics_database.csv", "r") as archivo_csv:
    lector = csv.reader(archivo_csv)
    paises_exportadores_data = []
    paises_importadores_data = []

    i = 0
    for linea in lector:
        if linea[1] == 'Exports': 
            paises_exportadores_data.append([linea[2], linea[3]])
          
        elif linea[1] == 'Imports':    
            paises_importadores_data.append([linea[2], linea[3]])
    
    paises_exportadores = []

    for pais in paises_exportadores_data:
        if pais not in paises_exportadores:
            paises_exportadores.append(pais)   
    
    paises_importadores = []

    for pais in paises_importadores_data:
        if pais not in paises_importadores:
            paises_importadores.append(pais)

   



In [55]:
with open("synergy_logistics_database.csv", "r") as archivo_csv:
    lector = csv.reader(archivo_csv)
    paises_exportadores_suma = []

    precios = []
    for linea in lector:
        if linea[9] != 'total_value': 
            precios.append(linea[9])

    total = 0
    for pais in paises_exportadores:
        suma = 0
        i = 0
        for linea in paises_exportadores_data:
            if pais[0] == linea[0] and pais[1] == linea[1]:
                suma += int(precios[i])
        i+=1
        
        paises_exportadores_suma.append([suma, pais])
        total += suma
    paises_exportadores_suma.sort(reverse=True)    
    
    print(total, "Valor de exportaciones total" )

    ochenta_porciento = 406771200000
    total = 0
    paises_exportadores_ochenta = []
    for pais in paises_exportadores_suma:   
        total +=pais[0]
        if (total <= ochenta_porciento):
            paises_exportadores_ochenta.append(pais)

    print("=================Paises exportadores que representan el 80%======================")
    for pais in paises_exportadores_ochenta:
        print(pais)



508464000000 Valor de exportaciones
=================Paises exportadores que representan el 80%======================
[16401000000, ['South Korea', 'Vietnam']]
[14421000000, ['Netherlands', 'Belgium']]
[14388000000, ['USA', 'Netherlands']]
[10890000000, ['China', 'Mexico']]
[10098000000, ['Japan', 'Brazil']]
[9867000000, ['Germany', 'France']]
[9207000000, ['South Korea', 'Japan']]
[9009000000, ['Australia', 'Singapore']]
[8613000000, ['Canada', 'Mexico']]
[8250000000, ['China', 'Spain']]
[8217000000, ['Mexico', 'New Zealand']]
[7887000000, ['Netherlands', 'France']]
[7788000000, ['Germany', 'USA']]
[7557000000, ['United Kingdom', 'Belgium']]
[7359000000, ['France', 'Belgium']]
[7227000000, ['Russia', 'Belorussia']]
[7062000000, ['India', 'United Arab Emirates']]
[6963000000, ['USA', 'Belgium']]
[6732000000, ['China', 'Rusia']]
[6699000000, ['Japan', 'Canada']]
[6600000000, ['Australia', 'Thailand']]
[6468000000, ['Japan', 'China']]
[6402000000, ['USA', 'Mexico']]
[6237000000, ['China'

In [59]:
with open("synergy_logistics_database.csv", "r") as archivo_csv:
    lector = csv.reader(archivo_csv)
    paises_importadores_suma = []

    precios = []
    for linea in lector:
        if linea[9] != 'total_value': 
            precios.append(linea[9])

    total = 0
    for pais in paises_importadores:
        suma = 0
        i = 0
        for linea in paises_importadores_data:
            if pais[0] == linea[0] and pais[1] == linea[1]:
                suma += int(precios[i])
        i+=1
        
        paises_importadores_suma.append([suma, pais])
        total += suma
    paises_importadores_suma.sort(reverse=True)    
    
    print(total, "Valor de importaciones total" )

    ochenta_porciento = 96307200000
    total = 0
    paises_importadores_ochenta = []
    for pais in paises_importadores_suma:   
        total +=pais[0]
        if (total <= ochenta_porciento):
            paises_importadores_ochenta.append(pais)

    print("=================Paises importadores que representan el 80%======================")
    for pais in paises_importadores_ochenta:
        print(pais)

120384000000 Valor de importaciones
=================Paises importadores que representan el 80%======================
[9009000000, ['Singapore', 'Thailand']]
[7689000000, ['Germany', 'China']]
[6930000000, ['China', 'Japan']]
[6798000000, ['Japan', 'Mexico']]
[6600000000, ['China', 'Thailand']]
[6435000000, ['Malaysia', 'Thailand']]
[4686000000, ['Spain', 'Germany']]
[4026000000, ['Mexico', 'USA']]
[3762000000, ['China', 'United Arab Emirates']]
[3729000000, ['Brazil', 'China']]
[3597000000, ['USA', 'Thailand']]
[3366000000, ['Australia', 'Japan']]
[3300000000, ['Mexico', 'China']]
[2838000000, ['France', 'Germany']]
[2739000000, ['France', 'Poland']]
[2673000000, ['USA', 'Canada']]
[2508000000, ['Japan', 'United Arab Emirates']]
[2310000000, ['Japan', 'Canada']]
[2310000000, ['Germany', 'Mexico']]
[2178000000, ['South Korea', 'Mexico']]
[2145000000, ['Mexico', 'Japan']]
[1914000000, ['United Kingdom', 'Canada']]
[1914000000, ['USA', 'Germany']]
[1914000000, ['Italy', 'Poland']]


Con está opción, pese a ser buena en terminos del valor de las exportaciones e importaciones. Se estarían dejando de lado completamente algunas rutas de exportación e importación, lo cual puede afectar negativamente a la contraparte de esta. 
Por poner un ejemplo, si se abandona la ruta de Fracia-Canada en importaciones por no ser parte de ese 80% más significativo, entoces, la ruta Canada-Francia en exportaciones, podría verse afectada de manera negativa. 

## 2. Conclución

#### Luego de explorar las soluciones antes mencionadas, la mejor opción es la segunda, pues presenta una gran flexibilidad de implementación. Ya que en lugar de elimiar el transporte por aire, sería mejor enfocarse en el transporte por mar, tierra y aire, además deja mucho terreno para buscar alternativas viables para transporte aereo, como lo pueden ser alianzas con aerolineas.